packages/sanity/src/core/form/store/types/nodes.ts

```diff
@@ -11,14 +12,50 @@ import {
 
 import {type FormNodePresence} from '../../../presence'
 import {type ObjectItem} from '../../types'
+import {type ProvenanceDiffAnnotation} from './diff'
 import {type FormFieldGroup} from './fieldGroup'
 import {type ArrayOfObjectsMember, type ArrayOfPrimitivesMember, type ObjectMember} from './members'
 
+/**
+ * @public
+ */
+export type ComputeDiff<Annotation> = (value: unknown) => Diff<Annotation>
+
+/**
+ * Props that encapsulate changes in the node's value.
+ *
+ * @public
+ */
+export interface NodeDiffProps<Annotation> {
+  /**
+   * The diff between the current value and the value the node is being compared to.
+   *
+   * This is marked as unstable because the API may need to evolve as we iterate on the advanced
+   * version control functionality. It will be stabilised when that project has matured.
```

Comment:
I think we can also add a `@beta` here or `@alpha` here to indicate the unstable nature of it. We don't use these consistently so I don't have a huge opinion on it but it might be a clearer and more immediate indicator of the danger 🤷